portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,24 +14,90 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import com.google.gson.Gson;
 import java.util.ArrayList;
 import java.util.Arrays;
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private final static int DEFAULT_NUMBER_OF_COMMENTS = 10;
+  private DatastoreService datastore;
+  private int maxComments;
+  @Override
+  public void init() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    maxComments = DEFAULT_NUMBER_OF_COMMENTS;
+  }
   
-  ArrayList<String> comments = new ArrayList<>(Arrays.asList("Yo this website is soooooooo cool", "That's cap bro this website is trash", "nah bro youre trash")); 
-  
-  @Override public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+  @Override 
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    maxComments = getMaxCommentChoice(request);
+    
+    ArrayList<Comment> comments = new ArrayList<>();
```

Comment:
If you know the number of elements beforehand, you can pass it to the constructor so it allocates that many.